packages/cache/src/internal/tar.ts

```diff
@@ -49,18 +54,28 @@ function getWorkingDirectory(): string {
 }
 
 // Common function for extractTar and listTar to get the compression method
-function getCompressionProgram(compressionMethod: CompressionMethod): string[] {
+async function getCompressionProgram(
+  compressionMethod: CompressionMethod
+): Promise<string[]> {
   // -d: Decompress.
   // unzstd is equivalent to 'zstd -d'
   // --long=#: Enables long distance matching with # bits. Maximum is 30 (1GB) on 32-bit OS and 31 (2GB) on 64-bit.
   // Using 30 here because we also support 32-bit self-hosted runners.
+  const tarPath = await getTarPath([])
```

Comment:
getTarPath is only used for getting to know if `SystemTarPathOnWindows` is used or not. This can be simplified so that we do not need to fit `args` array.
For example either:
1. Create a separate function to just get tar path
2. Or move settings the args to `getCompressionProgram` where anyways we are setting more arguments

Reply:
args set in getTarPath are only the ones related to particular os which also helps in deciding which tar to use.

Reply:
In that case we can break into two function - one to get tar path and another to to get archive command.